data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/AbstractPatternBasedMethod.java

```diff
@@ -489,6 +508,18 @@ private boolean attemptProjection(@NonNull MethodMatchContext matchContext, @Non
             if (pp == null) {
                 pp = entity.getIdOrVersionPropertyByName(propertyName);
             }
+            SourcePersistentEntity association = null;
+            if (pp == null && propertyName.endsWith(DtoJoinPathsParser.CAPITALIZED_ID)) {
```

Comment:
Seems an arbitrary limitation this `endsWith` check. I think `entity.getPath(propertyName)` should be called since if that returns a path you have a query on any property of the association.